plugins/out_s3/s3.c

```diff
@@ -119,6 +120,38 @@ static char *mock_error_response(char *error_env_var)
     return NULL;
 }
 
+static void s3_retry_warn (struct flb_s3 *ctx, char tag, char input_name, time_t create_time, bool less_than_limit){
+    struct tm now_time;
+    char create_time_str[20];
+    strftime(create_time, 20, "%Y-%m-%d %H:%M:%S", localtime_r(&create_time, &now_time));
+            if(less_than_limit) {
+                flb_plg_warn(ctx->ins,
+                             "failed to flush chunk tag=%s, create_time=%s"
+                             "retry issued: input=%s > output=%s (out_id=%d)",
+                             tag, create_time_str, input_name, ctx->ins->name, ctx->ins->id);
+            } else {
+                flb_plg_warn(ctx->ins,
+                             "chunk tag=%s, create_time=%s cannot be retried: input=%s > output=%s",
+                             tag, create_time_str, input_name, ctx->ins->name);
+            }
+} 
+
+static void s3_retry_warn_ctx (struct flb_s3 *ctx, char tag, time_t create_time, bool less_than_limit){
```

Comment:
This function and `s3_retry_warn` are almost exactly the same, hence why I was thinking they could all be one function. I think this makes it simpler.